### ISSUE TITLE
webview_flutter-1.0.7/ios/Classes/JavaScriptChannelHandler.h:5:9: fatal error: 'Flutter/Flutter.h' file not found     #import <Flutter/Flutter.h>

### ISSUE THREAD

**Issue by coderGirl9796** [flutter/flutter#80278]
## Steps to Reproduce

flutter run

## Logs

<details>
<summary>Logs</summary>

```
      Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    Command CompileSwift failed with a nonzero exit code
    In file included from /Users/bhumibachudasama/FlutterSdk/flutter/.pub-cache/hosted/pub.dartlang.org/webview_flutter-1.0.7/ios/Classes/JavaScriptChannelHandler.m:5:
    /Users/bhumibachudasama/FlutterSdk/flutter/.pub-cache/hosted/pub.dartlang.org/webview_flutter-1.0.7/ios/Classes/JavaScriptChannelHandler.h:5:9: fatal error: 'Flutter/Flutter.h' file not found
    #import <Flutter/Flutter.h>
            ^~~~~~~~~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Planning build
    note: Constructing build description
-->

```

```
[✓] Flutter (Channel stable, 2.0.4, on Mac OS X 10.15.7 19H15 darwin-x64, locale en-GB)
    • Flutter version 2.0.4 at /Users/bhumibachudasama/FlutterSdk/flutter
    • Framework revision b1395592de (11 days ago), 2021-04-01 14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/bhumibachudasama/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3, Build version 11C29
    ! Xcode 11.3.0 out of date (12.0.1 is recommended).
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 51.0.1
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.54.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (3 available)
    • sdk gphone x86 arm (mobile) • emulator-5554                        • android-x86    • Android 11 (API 30) (emulator)
    • iPhone 11 Pro Max (mobile)  • F8F09C34-1306-456F-B4C7-4A125710EEE7 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)
    • Chrome (web)                • chrome                               • web-javascript • Google Chrome 89.0.4389.114

! Doctor found issues in 1 category.
```

</details>

**Comment by TahaTesser** [comment]
Hi @coderGirl9796 
First make sure you have the latest `webview_flutter: ^2.0.4` (you're using `webview_flutter-1.0.7`) and then clean your project `flutter clean` and run the app.
If the problem persists, Can you please complete `flutter run --verbose` logs (logs will be really long so copy them in a text field and drop in the comments below)
Thank you

**Comment by justkawal** [comment]
@TahaTesser Same issue with 2.0.4

**Comment by TahaTesser** [comment]
@justkawal 
Delete `Podspec`, `Podfile` and `Podfile.lock files from `ios` folder and then `lfutter pub get` and `flutter run -v`

**Comment by justkawal** [comment]
@TahaTesser 
I tried the same but it didn't worked on Mac M1, as after it new error was showing 9.0 platform version was not supported because simulator was on iOS 14.4.

After I changed platform to 14.4 then the same error started again.
In case of emergency building, I had to cherry pick flutter: 1.22.6 from the flutter git and then it builds successfully.

**Comment by TahaTesser** [comment]
@justkawal 
Try 
`flutter pub cache repair` or delete `flutter/.pub-cache`

**Comment by justkawal** [comment]
@TahaTesser Still the same error.

Flutter Doctor :

```
[✓] Flutter (Channel stable, 2.0.4, on macOS 11.2.3 20D91 darwin-arm, locale en-IN)
    • Flutter version 2.0.4 at /Users/kawal/flutter
    • Framework revision b1395592de (2 weeks ago), 2021-04-01 14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/kawal/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Library/Java/JavaVirtualMachines/jdk1.8.0_281.jdk/Contents/Home/bin/java
    • Java version Java(TM) SE Runtime Environment (build 1.8.0_281-b09)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] Connected device (3 available)
    • iPhone 11 (mobile) • 4B19356E-B*************6A0DF9CA • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)    • macos                                • darwin-arm64   • macOS 11.2.3 20D91 darwin-arm
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 89.0.4389.128

! Doctor found issues in 1 category.
```
Pod File: 
```
# Uncomment this line to define a global platform for your project
platform :ios, '9.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
   target.build_configurations.each do |config|
    config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '9.0'
   end
  end
end
```
Error:
```
Launching lib/main.dart on iPhone 11 in debug mode...
Running pod install...                                              7.7s
Running Xcode build...                                                  
Xcode build done.                                            7.2s
Failed to build iOS app
Error output from Xcode build:
↳
    objc[27416]: Class AMSupportURLConnectionDelegate is implemented in both ?? (0x201af8188) and ?? (0x1168b02b8). One of the two will be used.
    Which one is undefined.
    objc[27416]: Class AMSupportURLSession is implemented in both ?? (0x201af81d8) and ?? (0x1168b0308). One of the two will be used. Which one is
    undefined.
    ** BUILD FAILED **


Xcode's output:
↳
    Command CompileSwift failed with a nonzero exit code
    In file included from /Users/kawal/.pub-cache/hosted/pub.dartlang.org/webview_flutter-2.0.4/ios/Classes/JavaScriptChannelHandler.m:5:
    /Users/kawal/.pub-cache/hosted/pub.dartlang.org/webview_flutter-2.0.4/ios/Classes/JavaScriptChannelHandler.h:5:9: fatal error:
    'Flutter/Flutter.h' file not found
    #import <Flutter/Flutter.h>
            ^~~~~~~~~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

Could not build the application for the simulator.
Error launching application on iPhone 11.
```

**Comment by justkawal** [comment]
@TahaTesser After some tweaks and major changes in versions as nullOk is removed from Flutter's recent version,
It is now running.

You can close the issue,

Thanks for help. @TahaTesser

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by justkawal** [comment]
@pedromassangocode 
🤔 Are you seriously reading my last comment ?? 🤣 

The problem is already resolved @pedromassangocode
